rio_stac/stac.py

```diff
@@ -10,13 +10,15 @@
 import numpy
 import pystac
 import rasterio
+from pystac.utils import str_to_datetime
 from rasterio import transform, warp
 from rasterio.features import bounds as feature_bounds
 from rasterio.io import DatasetReader, DatasetWriter, MemoryFile
 from rasterio.vrt import WarpedVRT
 
 PROJECTION_EXT_VERSION = "v1.0.0"
 RASTER_EXT_VERSION = "v1.1.0"
+EO_EXT_VERSION = "v1.0.0"
```

Comment:
Since you have a PySTAC dependency, you _could_ DRY by pulling the extension version from PySTAC, e.g. https://pystac.readthedocs.io/en/stable/api/extensions/eo.html#pystac.extensions.eo.EOExtension.get_schema_uri.

Reply:
I though about relying on pystac for this but I prefer having the full control here. e.g if something was to change in pystac extension this could maybe (in theory I hope not) lead to breaking change here. 

if we were to pin pystac to a specific version maybe this will be safer but I think for the time being this is the best way to make sure the content we built is matching the extension specification

Reply:
I think your approach makes sense. The extension story in PySTAC is a bit messy (e.g. https://github.com/stac-utils/pystac/issues/448).